discord/guild.py

```diff
@@ -764,6 +766,22 @@ def get_emoji(self, emoji_id: int, /) -> Optional[Emoji]:
         if emoji and emoji.guild == self:
             return emoji
         return None
+    
+    def get_guild_shop(self) -> Optional[GuildShop]:
+        """Returns the guild shop.
+
+        .. versionadded:: 2.4
+
+        Returns
+        -------
+        Optional[:class:`GuildShop`]
+            The returned Guild Shop or ``None`` if no shop is enabled.
+        """
+
+        shop = self._state.http.get_guild_shop(self.id)
+        if shop:
+            return shop # type: ignore
```

Comment:
Couple things:
(Note this assumes the issues outlined above are fixed, as in reality at the moment `shop` is a coroutine)

- This shouldn't directly return the dictionary parsed from the response.
  - Instead it should be converting the response object into a class representation.
  - The return type annotation should be updated to reflect this.
 -  The `type: ignore` comment should be removed.

Reply:
The type ignore is just to prevent MyPy syntax highlighting

Reply:
Fixed that also in the new commits